monkey/monkey_island/cc/services/config_schema_service.py

```diff
@@ -0,0 +1,79 @@
+from copy import deepcopy
+from typing import Any, Dict, Sequence, Tuple
+
+from common.agent_configuration import AgentConfiguration, PluginConfiguration
+from common.agent_plugins import AgentPlugin, AgentPluginType
+from monkey_island.cc.repositories import IAgentPluginRepository
+
+
+class ConfigSchemaService:
+    """
+    A service for retrieving the agent configuration schema.
+    """
+
+    def __init__(
+        self,
+        agent_plugin_repository: IAgentPluginRepository,
+    ):
+        self._agent_plugin_repository = agent_plugin_repository
+
+    def get_schema(self) -> Dict[str, Any]:
+        """
+        Get the schema.
+
+        :raises RuntimeError: If the schema could not be retrieved
+        """
+        try:
+            agent_config_schema = deepcopy(AgentConfiguration.schema())
```

Comment:
Where does it get defaults from?

Reply:
Take a read here: https://docs.pydantic.dev/usage/schema/

Reply:
I know, what I'm saying is that defaults, titles and descriptions are not defined in `AgentConfiguration`. titles and descriptions are generated, but done so poorly